public/js/goalCreation.js

```diff
@@ -0,0 +1,135 @@
+document.addEventListener("DOMContentLoaded", function() {
+    const goalForm = document.getElementById("goal_form");
+    const addGoalBtn = document.getElementById("add_goal_btn");
+    let goalsCount = 0;
+
+    goalForm.addEventListener("submit", function(event) {
+        event.preventDefault(); // Prevent form submission
+
+        // Validate goal fields
+        const goalName = document.getElementById("goal_name").value;
+        const goalDescription = document.getElementById("goal_description").value;
+        const goalAmountInput = document.getElementById("goal_amount");
+        const goalAmount = parseFloat(goalAmountInput.value.replace(/\$/g, '')); // remove $ sign before parsing
+        const startDate = document.getElementById("goal_start_date").value;
+        const endDate = document.getElementById("goal_end_date").value;
+
+        if (goalName.trim() === "") {
+            alert("Please enter the goal name.");
+            return;
+        }
+
+        if (isNaN(goalAmount) || goalAmount < 20) {
+            alert("Goal amount must be at least $20 and in a valid currency format.");
+            return;
+        }
+
+        const today = new Date().toISOString().split('T')[0];
+        if (startDate === "" || endDate === "" || startDate < today || startDate > endDate) {
+            alert("Please enter valid start and end dates.");
+            return;
+        }
+
+        // Format goal amount to currency with $0.00 format
+        goalAmountInput.value = goalAmount.toLocaleString('en-US', { style: 'currency', currency: 'USD', minimumFractionDigits: 2, maximumFractionDigits: 2 });
```

Comment:
This was what I was referencing when I commented in the goal creation html file. I think you are formatting the string here accordingly

Reply:
We can move forward with this